docs/src/pages/guides/migration-v4/migration-v4.md

```diff
@@ -19,32 +15,167 @@ We'll do our best to keep things easy to follow, and as sequential as possible,
+
+> 🛎 always create small commit on any changes to help the migration goes smoother.
+> Please check [Troubleshooting](#troubleshooting) section, if you encounter any issue or [create an issue](https://github.com/mui-org/material-ui/issues/new?assignees=&labels=status%3A+needs+triage&template=1.bug.md) with this title format `[Migration] Summary of your issue`.
+
+## **Update React & TypeScript version**
+
+- The minimum supported version of **React** was increased from v16.8.0 to v17.0.0.
+- The minimum supported version of **TypeScript** was increased from v3.2 to v3.5.
+
+  > We try to align with types released from [DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped) (i.e. packages published on npm under the `@types` namespace).
+  > We will not change the minimum supported version in a major version of Material-UI.
+  > However, we generally recommend to not use a TypeScript version older than the [lowest supported version of DefinitelyTyped](https://github.com/DefinitelyTyped/DefinitelyTyped#older-versions-of-typescript-33-and-earlier)
+
+**Note:** if your project includes these packages, please upgrade them to the `latest` version.
 
-The very first thing you will need to do is to update your dependencies.
+- `react-scripts`
+- `@types/react`
+- `@types/react-dom`
 
-### Update Material-UI version
+> ✅ Please make sure that your application is still **running** without error and **commit** the change before continuing the next step.
 
-You need to update your `package.json` to use the latest version of Material-UI and its peer dependencies.
+## **`ThemeProvider` Setup**
+
+Make sure that `MuiThemeProvider` is defined at the root of your application (even if you are using the **default theme**) and **NO** `useStyles` is called before `<MuiThemeProvider>`.
+
+```js
+import {
+  MuiThemeProvider,
+  createMuiTheme,
+  makeStyles,
+} from '@material-ui/core/styles';
+
+const theme = createMuiTheme();
+
+const useStyles = makeStyles((theme) => {
+  root: {
+    // some css that access to theme
+  }
+});
 
-```json
-"dependencies": {
-  "@emotion/react": "^11.0.0",
-  "@emotion/styled": "^11.0.0",
-  "@material-ui/core": "^5.0.0"
+function App() {
+  const classes = useStyles(); // ❌ If you have this, consider moving <ThemeProvider> to HOC and wrap the App
+  return <MuiThemeProvider theme={theme}>{children}</MuiThemeProvider>;
 }
 ```
 
-Or run
+> ✅ Please make sure that your application is still **running** without error and **commit** the change before continuing the next step.
+
+## **Update Material-UI version**
+
+To use the latest version of Material-UI.
 
 ```sh
 npm install @material-ui/core@next @emotion/react @emotion/styled
 
-or
-
+// or with `yarn`
 yarn add @material-ui/core@next @emotion/react @emotion/styled
 ```
 
+**Optional** if your project includes `@material-ui/icons` and/or `@material-ui/lab`, use the latest version of them.
+
+```sh
+npm install @material-ui/icons@next @material-ui/lab@next
```

Comment:
Should we include here also the emotion dependencies? And maybe link to the [Installation guide](https://deploy-preview-26948--material-ui.netlify.app/getting-started/installation/) in case they want to use styled-components.